### ISSUE TITLE
Default plugin search paths partially broken in alpha2

### ISSUE THREAD

**Issue by thokra1** [annulen/webkit#595]
It seems plugin support für NP Plugins is broken in the most recent releases.

Replacing TP5 with 5.212 in an otherwise unaltered project will break plugin loading and the infamous blue lego will show up in the plugin drawable.

**Comment by annulen** [comment]
Could you bisect what patch caused regression?

**Comment by annulen** [comment]
Also, on what platform? How do you install QtWebKit?

**Comment by thokra1** [comment]
I'll try to - maybe I can get around to a PR. Win10 x64, we used both the precompiled binaries you supplied in "Releases" and builds from source, all of which properly setup the corresponding CMake options. Since we're using CMake as well, we're incorporating WebKit directly via `find_package`.

**Comment by thokra1** [comment]
BTW: it's trivbial but do you need a PR for Debug and RelWithDebInfo CMake target modules? Were those maybe simply missing from the latest release? You provided debug binaries/PDBs but only the Release modules.

**Comment by annulen** [comment]
@Emdek says that Alpha 2 works fine on Win10 x64

**Comment by Emdek** [comment]
Yeah, works fine here, using this as test page:
http://copyboost.com/ad-examples/good-flash-ad-examples/

I would suggest (just in case, to be 1000% sure ;-)) that `QWebSettings::PluginsEnabled` is set to `true` and NPAPI variant of flash plugin is installed.
BTW, note that now there is also `QWebSettings::pluginSearchPaths()` and `QWebSettings::setPluginSearchPaths()`, you can try examining default paths or adding extra ones as last resort.

**Comment by thokra1** [comment]
@Emdek thanks for the hints on the search paths. 

It's strange, because all we're really doing is to set another PREFIX_PATH during config to select either TP5 or Alpha2 - the first resulting build works, the second doesn't. :D

I'll have to dig a little, I suppose.

**Comment by annulen** [comment]
Can you just replace dlls in existing build?

**Comment by thokra1** [comment]
@annulen Let me first confirm that I'm not doing something stupid. 😆  Swapping the runtime files was the first thing we opted for, but the remarks by @Emdek kinda make me doubt my assessment.

**Comment by thokra1** [comment]
So, after some investigating I stumbled upon this irregularity in the default search paths. I let the app print the paths right after entering `main` with 
```cpp
for (auto const& path : QWebSettings::pluginSearchPaths())
  std::cout << path.toStdString() << std::endl;
```

tp5
-
C:\LumoProjects\MB_3D-POS\bin\Plugins
C:\PFiles\Plugins
C:\Program Files\Windows Media Player
C:\WINDOWS\system32\macromed\Flash
C:\WINDOWS\system32\macromed\Shockwave 10

alpha2
-
C:\LumoProjects\MB_3D-POS\bi\Plugins
C:\PFiles\Plugins
C:\Program Files\Windows Media Player
C:\WINDOWS\system32\macromed\Flas
C:\WINDOWS\system32\macromed\Shockwave 1

Notice how the "n" is missing from "bin" and the "h" and the "0" are missing from their respective paths? 

So I put the pluing into C:\PFiles\Plugins and looky there, it works. So, my hypothesis is that there is truncation at play during path composition for alpha2.

**Comment by annulen** [comment]
"Flas" instead of "Flash" indicates that there is off-by-one error in 3d7087b27aa59d5885187dd79509c0919e9c9d43

**Comment by thokra1** [comment]
Cool! I updated the title to match the actual problem.

**Comment by annulen** [comment]
Fixed in 8368c860e2e7949dae175abb5bb87cd2e4bf8206
